### PR TITLE
updating change log after 3.6.2

### PR DESCRIPTION
**Pull request by cshanahan1**
(i wasn't sure if i needed to add a new 3.6.3 section, or if the existing 3.7 will be the next release, let me know if i should remove this new section)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -82,7 +82,28 @@ Specviz2d
...
+Mosviz
+^^^^^^
+
+Specviz
+^^^^^^^
+
+Specviz2d
+^^^^^^^^^
+
+3.6.2 (08-25-2023)
```
Comment (pllim) [inline CHANGES.rst:106]:
```suggestion
3.6.2 (2023-08-25)
```

**Review by rosteen — APPROVED** [review]
This works, we can always delete the 3.6.3 section if we release 3.7 without doing another bugfix beforehand.

File: CHANGES.rst (hunk 1 above)
Comment (rosteen) [reply CHANGES.rst:106]: Good catch! I didn't even notice 🙈

**Comment by pllim** [comment]
No need to wait for CI. Thanks!

**Pull request merged**